io/plugins/eu.esdihumboldt.hale.io.jdbc.mssql/src/eu/esdihumboldt/hale/io/jdbc/mssql/MsSqlGeometries.java

```diff
@@ -0,0 +1,231 @@
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.hale.io.jdbc.mssql;
+
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+import org.geotools.geometry.jts.JTS;
+import org.geotools.referencing.CRS;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.MathTransform;
+
+import com.google.common.io.BaseEncoding;
+import com.microsoft.sqlserver.jdbc.SQLServerConnection;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.io.WKTReader;
+
+import de.fhg.igd.slf4jplus.ALogger;
+import de.fhg.igd.slf4jplus.ALoggerFactory;
+import eu.esdihumboldt.hale.common.instance.geometry.DefaultGeometryProperty;
+import eu.esdihumboldt.hale.common.instance.geometry.impl.CodeDefinition;
+import eu.esdihumboldt.hale.common.instance.geometry.impl.WKTDefinition;
+import eu.esdihumboldt.hale.common.schema.geometry.CRSDefinition;
+import eu.esdihumboldt.hale.common.schema.geometry.GeometryProperty;
+import eu.esdihumboldt.hale.common.schema.model.TypeDefinition;
+import eu.esdihumboldt.hale.common.schema.model.impl.DefaultTypeDefinition;
+import eu.esdihumboldt.hale.io.jdbc.GeometryAdvisor;
+import eu.esdihumboldt.hale.io.jdbc.constraints.GeometryMetadata;
+import eu.esdihumboldt.hale.io.jdbc.mssql.util.SRSUtil;
+import schemacrawler.schema.Column;
+import schemacrawler.schema.ColumnDataType;
+
+/**
+ * Geometry advisor for MSSQL database
+ * 
+ * @author Arun
+ */
+public class MsSqlGeometries implements GeometryAdvisor<SQLServerConnection> {
+
+	private static final ALogger log = ALoggerFactory.getLogger(MsSqlGeometries.class);
+
+	@Override
+	public Object convertGeometry(GeometryProperty<?> geom, TypeDefinition columnType,
+			SQLServerConnection connection) throws Exception {
+		if (geom instanceof DefaultGeometryProperty) {
+
+			// We need Column Data type
+			String columnDataType = columnType.getName().getLocalPart();
+
+			CoordinateReferenceSystem targetCRS = null;
+
+			Statement stmt = null;
+			ResultSet rs = null;
+			try {
+				int srId = 4326;
+
+				if (geom.getCRSDefinition() != null) {
+
+					// SRS Code
+					String srsName = CRS.toSRS(geom.getCRSDefinition().getCRS());
+
+					// Todo:: Do we really need transformation for 'geometry'
+					// columnDataType instances? because this type represents
+					// data in a Euclidean (flat) coordinate system.
+
+					// getting Axis order of geometry CRS
+					CRS.AxisOrder axisOrder = CRS.getAxisOrder(geom.getCRSDefinition().getCRS());
+					// if axis order is not x/y ordering then will need to
+					// transform geometry to target crs with longiture first
```

Comment:
That isn't exactly correct, because it may for instance be a projected reference system that does not use latitude/longitude deegrees at all.
Anyway, it doesn't hurt to do the transformation.
A better check might be to just compare the axis order with that of the target CRS - if it is the same, no transformation needs to be done.